transport/src/main/java/io/netty/channel/DefaultChannelPromise.java

```diff
@@ -37,7 +39,7 @@
      *        the {@link Channel} associated with this future
      */
     public DefaultChannelPromise(Channel channel) {
-        this.channel = channel;
+        this.channel = checkNotNull(channel, "channel");
     }
 
     /**
```

Comment:
does the constructor below also need to have a null check?

Reply:
@Scottmitch I thought about this but we pass in null explicit in `Bootstrap` if `ChannelFactory` throws. We could use a "dummy" `Channel` here as replacement of null but not sure this is better. Thoughts ?

Reply:
Seems strange to have once constructor allow null and the other not allow null in this scenario. The case of "no channel" does seem exceptional and a "dummy" channel seems reasonable here to make the interface contract more clear.

Reply:
done... PTAL again